File: attribute-compiler/src/main/java/pyxis/uzuki/live/attribute/parser/compiler/model/AttrStringModel.java
package pyxis.uzuki.live.attribute.parser.compiler.model;

import javax.lang.model.element.VariableElement;

import pyxis.uzuki.live.attribute.parser.annotation.AttrString;

/**
 * AttributesParser
 * Class: AttrIntModel
 * Created by Pyxis on 3/4/18.
 * <p>
 * Description:
 */

public class AttrStringModel extends BaseAttrModel {
    private String mSource;

    public AttrStringModel(VariableElement element) {
        super(element);

        AttrString annotation = element.getAnnotation(AttrString.class);

        mSource = annotation.value();

        findEnclosingClass();
    }

    public String getSource() {
        return mSource;
    }
}

File: attribute-parser/src/main/java/pyxis/uzuki/live/attribute/parser/model/AttrStringModel.kt
package pyxis.uzuki.live.attribute.parser.model

import pyxis.uzuki.live.attribute.parser.annotation.AttrString
import javax.lang.model.element.VariableElement

/**
 * AttributesParser
 * Class: AttrIntModel
 * Created by Pyxis on 3/4/18.
 *
 *
 * Description:
 */

class AttrStringModel(element: VariableElement) : BaseAttrModel(element) {
    val source: String

    init {

        val annotation = element.getAnnotation(AttrString::class.java)

        source = annotation.value

        findEnclosingClass()
    }
}